### PR TITLE
Fixing order of exported inputs in NLP models

### PR DESCRIPTION
**Pull request by borisfom**
Signed-off-by: Boris Fomitchev <bfomitchev@nvidia.com>

# What does this PR do ?

Fixing remaining issues with exported order of inputs of NLP models:
* Except Punct&Capital, exported order of inputs of BERT networks was wrong
* There were discrepancies between Bert and Roberta/CamamBert forward() argument orders in nlp/modules/common/huggingface

Also cleaned up unused input_types()/output_types() overloads

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/text_classification/text_classification_model.py [hunk 1]
```diff
@@ -104,7 +96,7 @@ def create_loss_module(self):
                 self.loss = CrossEntropyLoss()
 
     @typecheck()
-    def forward(self, input_ids, token_type_ids, attention_mask):
+    def forward(self, attention_mask, input_ids, token_type_ids):
```
Comment (ramanathan831) [inline nemo/collections/nlp/models/text_classification/text_classification_model.py]: input_ids, attention_mask, token_type_ids

Comment (borisfom) [reply nemo/collections/nlp/models/text_classification/text_classification_model.py]: Done.

**Review by ramanathan831 — APPROVED** [review]

File: nemo/collections/asr/modules/rnnt.py [hunk 2]
```diff
@@ -208,7 +208,7 @@ def predict(
 
         Args:
             y: Optional torch tensor of shape [B, U] of dtype long which will be passed to the Embedding.
-                If None, creates a zero tensor of shape [B, 1, H] which mimics output of pad-token on Embedding.
+                If None, creates a zero tensor of shape [B, 1, H] which mimics output of pad-token on EmbeddiNg.
```
Comment (jarivk) [inline nemo/collections/asr/modules/rnnt.py:211]: accidental change?

File: nemo/collections/nlp/modules/common/bert_module.py [hunk 3]
```diff
@@ -79,8 +79,12 @@ def input_example(self, max_batch=1, max_dim=256):
         """
         sample = next(self.parameters())
         sz = (max_batch, max_dim)
-        input_ids = torch.randint(low=0, high=2048, size=sz, device=sample.device)
+        input_ids = torch.randint(low=0, high=max_dim - 1, size=sz, device=sample.device)
```
Comment (jarivk) [inline nemo/collections/nlp/modules/common/bert_module.py:82]: Any reason for clamping high value to max_dim-1?
Shouldn't  `high` value depend on the size of the vocabulary?

File: nemo/utils/export_utils.py [hunk 4]
```diff
@@ -100,23 +100,24 @@ def parse_input_example(input_example):
...
 
-def to_onnxrt_input(input_names, input_list, input_dict):
+def to_onnxrt_input(input_names, input_dict, input_list):
     odict = {}
-    for k, v in input_dict.items():
-        odict[k] = v.cpu().numpy()
-    for i, input in enumerate(input_list):
-        if type(input) in (list, tuple):
-            odict[input_names[i]] = tuple([ip.cpu().numpy() for ip in input])
+    for k in reversed(input_names):
```
Comment (jarivk) [inline nemo/utils/export_utils.py:105]: any reason for using reversed?

**Comment by borisfom** [comment]
Included in #3893

File: nemo/utils/export_utils.py (hunk 4 above)
Comment (borisfom) [reply nemo/utils/export_utils.py:105]: For the theoretical use case when both input_list and input_dict are present - that would mean, by export() API rules,  the dict is the last, so we need to start from the back.

**Pull request closed without merging**